### PR TITLE
Added basic Hexgrid and basic Hex functionality

### PR DESCRIPTION
**Pull request by lokiy999**
This PR adds a function in the Grid class to generate a Hexgrid of size X, Z. The Hex class has a functionality to calculate the neighbors.

### CODE REVIEW

**Review by Westlanderz — CHANGES_REQUESTED** [review]
There are some codingstyle changes that need to be done, code in general looks good.

File: Assets/Scripts/Hex.cs [hunk 1]
```diff
@@ -0,0 +1,80 @@
...
+		} else {
+			return this.Add(Hex.directionsOdd[direction]);
+		}
+	}
+
+	// Print out all neighbors of a Hex.
+	public Hex[] Neighbors() {
+		Hex[] hexNeighbors = new Hex[6];
+		for(int x = 0; x < 6; x++) {
+			try {
```
[the remark this answers is not in the document]
Comment (lokiy999) [reply Assets/Scripts/Hex.cs:51]: I've changed the Try-Catch with if-else loops to reduce garbage collection.

File: Assets/Scripts/Hex.cs [hunk 2]
```diff
@@ -0,0 +1,80 @@
...
+	}
+
+	// Directions to calculate neighbors if the row is odd.
+	static public List<Hex> directionsOdd = new List<Hex>{
+		new Hex(-1, -1),
+		new Hex(-1, 0),
+		new Hex(-1, 1),
+		new Hex(0, -1),
+		new Hex(0, 1),
+		new Hex(1, 0)};
```
[the remark this answers is not in the document]
Comment (lokiy999) [reply Assets/Scripts/Hex.cs:70]: The closing braces are now below the Lists.

File: Assets/Scripts/GridLeon.cs [hunk 3]
```diff
@@ -0,0 +1,76 @@
...
+
+public class GridLeon : MonoBehaviour {
+	public Transform hexPrefab;
+	public Transform hexPrefab2;
+
+	public int gridWidth = 11;
+	public int gridHeight = 11;
+
+	// These values have been carefully chosen after a lot of 'calculating'.
+	float hexWidth = 1.732f;
```
[the remark this answers is not in the document]
Comment (lokiy999) [reply Assets/Scripts/GridLeon.cs:11]: The private members hexWidth and HexHeight are now private and have a trailing _ attached to them. I also added private to startPos and added a trailing _.

File: Assets/Scripts/Hex.cs [hunk 4]
```diff
@@ -0,0 +1,80 @@
+using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+
+public class Hex : MonoBehaviour {
+	// Unity Location of the X and Z position.
+	private float unityPosX, unityPosZ;
```
[the remark this answers is not in the document]
Comment (lokiy999) [reply Assets/Scripts/Hex.cs:7]: I've added a trialing _ to unityPosX, unityposZ and rowIsEven.

**Review by Westlanderz — APPROVED** [review]
LGTM

**Pull request merged**